src/App.tsx

```diff
@@ -1,10 +1,50 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { FilterTypes } from './types/FilterTypes';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodos, setSelectedTodos] = useState<FilterTypes>(
+    FilterTypes.All,
+  );
+  const [isErrorHidden, setIsErrorHidden] = useState(true);
+
+  const areTodosExist = !!todos?.length;
+  const notCompletedTodosCount = todos?.filter(todo => !todo.completed).length;
+
+  function filterTodos(initTodos: Todo[], filterType: FilterTypes) {
+    switch (filterType) {
+      case FilterTypes.Active:
+        return initTodos.filter(todo => !todo.completed);
+      case FilterTypes.Completed:
+        return initTodos.filter(todo => todo.completed);
+      default:
+        return initTodos;
+    }
+  }
+
+  const filteredTodos = filterTodos(todos, selectedTodos);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(error => {
+        setIsErrorHidden(false);
+        alert(error);
+      })
+      .finally(() => {
+        setTimeout(() => {
+          setIsErrorHidden(true);
+        }, 3000);
```

Comment:
If you don't have an error so you don't need to reset it, and I suggest you to store in a state an error message instead, as you should show different errors in different cases. You can create a handler that sets an error and then clean it in 3 seconds, it will help you to don't repeat it in the future. Something like this:
```typescript
const handleError = (errorMessage: string) => {
  setErrorMessage(errorMessage);

  setTimeout(() => {
    setErrorMessage('');
  }, 3000);
};
```